src/Publisher/FilesystemPublisher.php

```diff
@@ -174,8 +215,25 @@ protected function saveToPath($content, $filePath)
         // Move the temporary file to the desired location (prevents unlocked files from being read during write)
         $publishPath = $this->getDestPath() . DIRECTORY_SEPARATOR . $filePath;
         Filesystem::makeFolder(dirname($publishPath));
+        $successWithPublish = rename($temporaryPath, $publishPath);
+        if ($successWithPublish) {
+            if (FilesystemPublisher::config()->get('use_gzip_compression')) {
+                $publishPath = $this->compressFile($publishPath);
+            }
+        }
+
+        return file_exists($publishPath);
+    }
+
+    protected function compressFile(string $publishPath): string
+    {
+        //we keep the html file for now ... so use second parameter to achieve this.
+        $publishPathGZipped = $publishPath . '.gz';
+        exec('rm ' . $publishPathGZipped . ' -rf');
+        exec('gzip ' . $publishPath);
```

Comment:
What if gzip is unavailable?

Why can't we gzip with PHP rather than using `exec`? This use of exec makes me a bit worried.